File: group_01/longestConsecutive.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int ans = 0;
        unordered_set<int> seen(nums.begin(), nums.end());
        for (int x : seen) {
            if (seen.contains(x - 1)) {
                continue;
            }
            int y = x + 1;
            while (seen.contains(y)) {
                y += 1;
            }
            ans = max(ans, y - x);
        }
        return ans;
    }
};

File: group_01/longestConsecutive.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        seen = set(nums)
        ans = 0

        for x in seen:
            if x - 1 in seen:
                continue
            y = x + 1
            while y in seen:
                y += 1
            ans = max(ans, y - x)

        return ans
